libs/api/src/lib/payments.ts

```diff
@@ -14,31 +14,41 @@ export function setupPaymentProvider(opts: WepublishServerOpts): Router {
     paymentProviderWebhookRouter
       .route(`/${paymentProvider.id}`)
       .all(paymentProvider.incomingRequestHandler, async (req, res, next) => {
-        await res.status(200).send() // respond immediately with 200 since webhook was received.
         logger('paymentProvider').info(
           'Received webhook from %s for paymentProvider %s',
           req.get('origin'),
           paymentProvider.id
         )
         try {
-          const paymentStatuses = await paymentProvider.webhookForPaymentIntent({req})
-          const context = await contextFromRequest(req, opts)
+          const webhookResponse = await paymentProvider.webhookForPaymentIntent({req})
+          switch (webhookResponse.status) {
+            case 200:
+              if (webhookResponse.paymentStates) {
+                const context = await contextFromRequest(req, opts)
 
-          for (const paymentStatus of paymentStatuses) {
-            // TODO: handle errors properly
-            await paymentProvider.updatePaymentWithIntentState({
-              intentState: paymentStatus,
-              paymentClient: context.prisma.payment,
-              paymentsByID: context.loaders.paymentsByID,
-              invoicesByID: context.loaders.invoicesByID,
-              subscriptionClient: context.prisma.subscription,
-              userClient: context.prisma.user,
-              invoiceClient: context.prisma.invoice,
-              subscriptionPeriodClient: context.prisma.subscriptionPeriod,
-              invoiceItemClient: context.prisma.invoiceItem
-            })
+                for (const paymentStatus of webhookResponse.paymentStates) {
+                  // TODO: handle errors properly
```

Comment:
does this comment still need to be here?

Reply:
I don't know tbh, I don't think we handle payment errors here at all. Another work to do imo.